rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageTestsBase.scala

```diff
@@ -97,28 +104,102 @@ abstract class InMemoryHotStoreTestsBase[F[_]]
   override def withTestSpace[S](f: (ST, T) => F[S]): S = {
     val branch = Branch("inmem")
 
+    implicit val patternCodec: Codec[Pattern] = StringExamples.implicits.patternSerialize.toCodec
+    implicit val stringCodec: Codec[String]   = StringExamples.implicits.stringSerialize.toCodec
+    implicit val stringCaptorCodec: Codec[StringsCaptor] =
+      StringExamples.implicits.stringClosureSerialize.toCodec
+
+    val dbDir: Path   = Files.createTempDirectory("rchain-storage-test-")
+    val mapSize: Long = 1024L * 1024L * 1024L
+
+    def storeConfig(name: String): StoreConfig =
+      StoreConfig(
+        Files.createDirectories(dbDir.resolve(name)).toString,
+        mapSize,
+        2,
+        2048,
+        List(EnvFlags.MDB_NOTLS)
+      )
+
+    val config = LMDBRSpaceStorageConfig(
+      storeConfig("cold"),
+      storeConfig("history"),
+      storeConfig("pointers"),
+      storeConfig("roots")
+    )
+
     run(for {
-      historyState <- Cell.refCell[F, Cache[String, Pattern, String, StringsCaptor]](
-                       Cache[String, Pattern, String, StringsCaptor]()
-                     )
-      historyReader = {
-        implicit val h = historyState
-        new History[F, String, Pattern, String, StringsCaptor]
-      }
+      historyRepository <- HistoryRepositoryInstances
+                            .lmdbRepository[F, String, Pattern, String, StringsCaptor](config)
       cache <- Cell.refCell[F, Cache[String, Pattern, String, StringsCaptor]](
                 Cache[String, Pattern, String, StringsCaptor]()
               )
       testStore = {
-        implicit val hr = historyReader
-        implicit val c  = cache
+        implicit val hr: HistoryReader[F, String, Pattern, String, StringsCaptor] =
+          historyRepository
+        implicit val c = cache
```

Comment:
again, the verbose implicits which could be avoided if you simply pass the reqiured parameters to the function inMem